runtime/compiler/optimizer/EscapeAnalysis.cpp

```diff
@@ -1076,6 +1076,23 @@ int32_t TR_EscapeAnalysis::performAnalysisOnce()
                traceMsg(comp(), "   Make [%p] non-local because we can't have locking when candidate escapes in cold blocks\n", candidate->_node);
             }
 
+         // Value type fields of objects created with a NEW bytecode must be initialized
+         // with their default values.  EA is not yet set up to perform such iniitialization,
+         // so remove the candidate from consideration.
+         if (candidate->_kind == TR::New)
+            {
+            TR_OpaqueClassBlock *clazz = (TR_OpaqueClassBlock *)candidate->_node->getFirstChild()->getSymbol()->getStaticSymbol()->getStaticAddress();
+
+            if (!TR::Compiler->cls.isZeroInitializable(clazz))
+               {
+               if (trace())
+                  traceMsg(comp(), "   Fail [%p] because the candidate is not zero initializable (that is, it has a field that is an unflattened value type)\n", candidate->_node);
```

Comment:
Just to clarify, "unflattened value type" means a value type that is not the reference project of the type?

Reply:
It means the fields of the value type field haven't been "inlined" into the class that contains that field.  I will reword this to make things more clear.

Reply:
Revised wording in commit https://github.com/eclipse/openj9/pull/9139/commits/d897b437a02626a45d5be525d10a50d192db14c1